src/Controller/UserController.php

```diff
@@ -64,4 +78,64 @@ public function register(UserDto $userDto): Response
 
         return new JsonResponse($savedDto, Response::HTTP_CREATED);
     }
+
+    /**
+     * @Route(path="/delete/{email}", methods={"DELETE"})
+     */
+    public function softDelete(string $email): Response
+    {
+        $user = $this->userRepository->findOneBy(['email' => $email]);
+
+        if (null === $user) {
+            $this->logger->info('Soft delete failed: no account for such email');
+
+            return new JsonResponse('No account associated with email', Response::HTTP_NOT_FOUND, [], true);
+        }
+        $this->eventDispatcher->dispatch(new UserSoftDeleteEvent($user), UserSoftDeleteEvent::NAME);
```

Comment:
you shouldn't actually create an event and process that in the event. you should delete the user here